pkg/provision/workspace/deployment.go

```diff
@@ -169,6 +169,35 @@ func ScaleDeploymentToZero(ctx context.Context, workspace *common.DevWorkspaceWi
 	return nil
 }
 
+// Returns an error message and a FailureReason if the workspace related pods are in an unrecoverable state, which may
+// have been caused  by an ignoredUnrecoverableEvent.
+// Otherwise, empty strings are returned for the error message and failure reason.
+func CheckForIgnoredWorkspacePodEvents(workspace *common.DevWorkspaceWithConfig, clusterAPI sync.ClusterAPI) (errMessage string, reason metrics.FailureReason) {
+	workspaceIDLabel := k8sclient.MatchingLabels{constants.DevWorkspaceIDLabel: workspace.Status.DevWorkspaceId}
+	// CheckPodsState returns either a message or error, not both.
+	errMsg, checkErr := status.CheckPodsState(workspace.Status.DevWorkspaceId, workspace.Namespace, workspaceIDLabel, []string{}, clusterAPI)
+	if checkErr != nil {
+		status := DeploymentProvisioningStatus{
+			ProvisioningStatus: ProvisioningStatus{
+				Err: checkErr,
+			},
+		}
```

Comment:
It's strange to build this struct only for the purpose of passing it into a metrics function. This function could just return `(string, error)`.